juspay/hyperswitch
# fix: do not allow duplicate organization name

## Type of Change



## Description

Fixes in this PR:
1. No two organizations can have same `organization_name`.
2. Don't allow unknown fields in organization create and update request body.
3. Made organization name required field during new organization creation.

### Additional Changes





## Motivation and Context



## How did you test it?

Manual
1. create an organization with the same name. (should fail with 4XX)
```
curl --location 'http://localhost:8080/v2/organization' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
    "organization_name": "my_org"
}'
```
Response:
```
{
    "error": {
        "type": "invalid_request",
        "message": "Organization with the given organization_name already exists",
        "code": "IR_38"
    }
}
```

2. send an invalid key-value pair in organization create and update call. (should fail with 4XX)
Create
```
curl --location 'http://localhost:8080/v2/organization' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
    "organization_name": "my_org",
    "unknown_key": "value"
}'
```
Update
```
curl --location --request PUT 'http://localhost:8080/v2/organization/org_DXkXLjRK6xIMBe6VWBnj' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
    "organization_name": "your_ org",
    "unknown_key": "value"
}'
```
Response:
```
{
    "error": "Json deserialize error: unknown field `unknown_key`, expected one of `organization_name`, `organization_details`, `metadata` at line 3 column 17"
}
```
3. Create a new organization without an organization name.(should fail with 4xx).
```
curl --location 'http://localhost:8080/v2/organization' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
    
    
}'
```
Response:
```
{
    "error": "Json deserialize error: missing field `organization_name` at line 4 column 1"
}
```


## Checklist
